src/helpers/yjs.js

```diff
@@ -21,27 +21,80 @@
  */
 
 import { encodeArrayBuffer, decodeArrayBuffer } from '../helpers/base64.js'
-import { Doc, encodeStateAsUpdate, applyUpdate } from 'yjs'
+import * as Y from 'yjs'
 import * as decoding from 'lib0/decoding.js'
+import * as encoding from 'lib0/encoding.js'
+import * as syncProtocol from 'y-protocols/sync'
+import { messageSync } from 'y-websocket'
 
 /**
+ * Get Document state encode as base64.
  *
- * @param {Doc} ydoc - encode state of this doc
+ * Used to store yjs state on the server.
+ * @param {Y.Doc} ydoc - encode state of this doc
  */
 export function getDocumentState(ydoc) {
-	const update = encodeStateAsUpdate(ydoc)
+	const update = Y.encodeStateAsUpdate(ydoc)
 	return encodeArrayBuffer(update)
 }
 
 /**
  *
- * @param {Doc} ydoc - apply state to this doc
+ * @param {Y.Doc} ydoc - apply state to this doc
  * @param {string} documentState - base64 encoded doc state
  * @param {object} origin - initiator object e.g. WebsocketProvider
  */
 export function applyDocumentState(ydoc, documentState, origin) {
 	const update = decodeArrayBuffer(documentState)
-	applyUpdate(ydoc, update, origin)
+	Y.applyUpdate(ydoc, update, origin)
+}
+
+/**
+ * Update message for everything in ydoc that is not in encodedBaseUpdate
+ *
+ * @param {Y.Doc} ydoc - encode state of this doc
+ * @param {string} encodedBaseUpdate - base64 encoded doc update to build upon
```

Comment:
Return type annotation would be good especially since undefined is an expected value